lib/tjbot.js

```diff
@@ -774,28 +786,78 @@ TJBot.prototype.read = function() {
     var self = this;
 
     return new Promise(function(resolve, reject) {
-        temp.open('tjbot', function(err, info) {
+        self.takePhoto().then(function(filePath) {
+            resolve(self.describePhotoText(filePath))
+        });
+    });
+}
+
+/*
+Describe text in photo by sending it to the Watson Visual Recognition Services.
+ */
+TJBot.prototype.describePhotoText = function(filePath) {
+    this._assertCapability('see');
+
+    // capture 'this' context
+    var self = this;
+
+    return new Promise(function(resolve, reject) {
+        winston.debug("sending image to Watson Visual Recognition to recognize text.");
+        var params = {
+            images_file: fs.createReadStream(filePath),
+            threshold: self.configuration.see.confidenceThreshold.text
+        };
+
+        self._visualRecognition.recognizeText(params, function(err, response) {
             if (err) {
-                reject("error: could not open temporary file for writing at path: " + info.path);
+                reject(err);
+            } else {
+                resolve(response);
             }
+        });
+    });
+}
 
-            self._captureImage(info.path).then(function(buffer) {
-                winston.debug("sending image to Watson Visual Recognition");
-                var params = {
-                    images_file: fs.createReadStream(info.path + ".jpg"),
-                    threshold: self.configuration.see.confidenceThreshold.text
-                };
 
-                self._visualRecognition.recognizeText(params, function(err, response) {
-                    if (err) {
-                        reject(err);
-                    } else {
-                        resolve(response);
-                    }
-                });
-            });
+/**
+ * Capture an image and save it in the given path. If no path is provided, it saves this file to a temp location
+ *
+ * @param {String} filePath The path at which to save the image.
+ *
+ * Returns the photo data in a Buffer.
+ */
+TJBot.prototype.takePhoto = function(filePath) {
+    this._assertCapability('see');
+
+    // capture 'this' context
+    var self = this;
+    var path = "";
+    var name = "";
+
+    // if no file path provided, save to temp location
+    if (filePath == null || filePath == "") {
+        filePath = temp.path({
+            prefix: 'tjbot'
         });
+    };
+
+    winston.debug("capturing image at path: " + filePath);
+    path = filePath.substring(0, filePath.lastIndexOf("/"));
+    name = filePath.substring(filePath.lastIndexOf("/") + 1);
+
+    this._camera.setOptions({
+        outputDir: path,
+        fileName: name
     });
+
+    return new Promise(function(resolve, reject) {
+        self._camera.takePhoto().then(function(photobuffer) {
+            resolve(path + "/" + name + "." + self._camera.getOptions().encoding);
+        }).catch((error) => {
```

Comment:
we should standardize on one way of doing callbacks:

**option A**
```
}).catch(function(error) {
   ...
```

**option B**
```
}).catch((error) => {
...
```

i think i used **option A** in all the previous code, i'd like this to be consistent